devicetypes/zooz/zooz-remote-switch-zen34.src/zooz-remote-switch-zen34.groovy

```diff
@@ -0,0 +1,321 @@
+	state.refreshAll = true
+	initialize()
+}
+
+def updated() {
+	logDebug "updated()..."
+	initialize()
+
+	if (pendingChanges) {
+		logForceWakeupMessage("The setting changes will be sent to the device the next time it wakes up.")
+	}
+}
+
+void initialize() {
+	state.debugLoggingEnabled = (safeToInt(settings?.debugOutput, 1) != 0)
+
+	refreshSyncStatus()
+
+	if (!device.currentValue("checkInterval")) {
+		sendEvent([name: "checkInterval", value: ((wakeUpInterval * 2) + (5 * 60)), displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID]])
+	}
+
+	if (!device.currentValue("supportedButtonValues")) {
+		sendEvent(name:"supportedButtonValues", value: supportedButtonValues.encodeAsJSON(), displayed:false)
+	}
+
+	if (!device.currentValue("numberOfButtons")) {
+		sendEvent(name:"numberOfButtons", value:1, displayed:false)
+	}
+
+	if (!device.currentValue("button")) {
+		sendButtonEvent("up")
+	}
+}
+
+def configure() {
+	logDebug "configure()..."
+	List<String> cmds = []
+
+	if (state.refreshAll || !device.currentValue("firmwareVersion")) {
+		cmds << secureCmd(zwave.versionV1.versionGet())
+	}
+
+	if (state.refreshAll || !device.currentValue("battery")) {
+		cmds << secureCmd(zwave.batteryV1.batteryGet())
+	}
+
+	state.refreshAll = false
+
+	if (state.wakeUpInterval != wakeUpInterval) {
+		cmds << secureCmd(zwave.wakeUpV2.wakeUpIntervalSet(seconds: wakeUpInterval, nodeid: zwaveHubNodeId))
+		cmds << secureCmd(zwave.wakeUpV2.wakeUpIntervalGet())
+	}
+
+	configParams.each { name, param ->
+		Integer storedVal = getStoredVal(name)
+		Integer settingVal = getSettingVal(name)
+		if (storedVal != settingVal) {
+			logDebug "Changing ${param.title}(#${param.num}) from ${storedVal} to ${settingVal}"
+			cmds << secureCmd(zwave.configurationV1.configurationSet(parameterNumber: param.num, size: param.size, scaledConfigurationValue: settingVal))
+			cmds << secureCmd(zwave.configurationV1.configurationGet(parameterNumber: param.num))
+		}
+	}
+	if (cmds) {
+		sendHubCommand(cmds, 500)
+	}
+}
+
+def ping() {
+	logDebug "ping()..."
+}
+
+def refresh() {
+	logDebug "refresh()..."
+	state.refreshAll = true
+
+	refreshSyncStatus()
+
+	logForceWakeupMessage("The next time the device wakes up, the sensor data will be requested.")
+}
+
+String secureCmd(cmd) {
+	if (zwaveInfo?.zw?.contains("s")) {
+		return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		return cmd.format()
+	}
+}
+
+def parse(String description) {
+	def cmd = zwave.parse(description, commandClassVersions)
+	if (cmd) {
+		zwaveEvent(cmd)
+	} else {
+		log.warn "Unable to parse: $description"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpIntervalReport cmd) {
+	logDebug "$cmd"
+	runIn(4, refreshSyncStatus)
+	state.wakeUpInterval = cmd.seconds
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	logDebug "Device Woke Up..."
+	runIn(4, refreshSyncStatus)
+	configure()
+}
```

Comment:
I don't see you ever sending a `NoMoreInformation` which technically violates the Z-Wave spec and probably will not do good things for your battery life.

I'd recommend you include it

Reply:
I've made the change to avoid being difficult, but there was no point in adding it.

SmartThings made a platform change about 3-6 months that causes it to suppress that command if it's specified in the handler and automatically send it 4 seconds after the last command is sent.

The first 2 lines in the screenshot below are from the original handler and the last 2 are from the version I just posted that sends NoMoreInformation, which proves that making this change has no impact on the behavior of the handler...

![image](https://user-images.githubusercontent.com/89558926/133538433-12467cc6-4e3f-48e0-a572-b81ee2ce5def.png)